src/content-scripts/components/AnnotationRequestForm/AnnotationRequestForm.tsx

```diff
@@ -54,21 +54,17 @@ interface AnnotationRequestFormState extends IAnnotationRequestFormData {
 export default class AnnotationRequestForm extends React.Component<Partial<AnnotationRequestFormProps>,
   Partial<AnnotationRequestFormState>> {
 
-  static getDerivedStateFromProps(nextProps, prevState) {
-    return {
+  constructor(props: AnnotationRequestFormProps) {
+    super(props);
+    this.state = {
       quote: '',
       comment: '',
-      ...nextProps.initialData,
+      ...props.initialData,
```

Comment:
Is constructor called every time props changes?
I am asking becuse such was current implementation and changing it will break how form behaves if it has been already opened when request annotation action is called for another quote.

Reply:
You're right, I forgot it can be updated (and not just mounted / unmounted)

Reply:
If we want to get rid of derived props I would follow this `key={id}` (or `key={quote}` in this case`) solution that we once considered and it also in the [article](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) you mentioned.

Reply:
(and no, it is not)

Reply:
I just thought that too so I'm going to do just that :)

Reply:
It is fixed now